db/collection_get.go

```diff
@@ -48,13 +48,19 @@ func (c *collection) get(
 	ctx context.Context,
 	txn datastore.Txn,
 	key core.PrimaryDataStoreKey,
+	fields []*client.FieldDescription,
```

Comment:
`Fetcher.Init` method takes as one of the parameters `[]*client.FieldDescription`. In this file `get` method used to pass `nil` to it, which mean all properties AFAIK. Now I let the caller of `get` decide which properties it want to fetch.

Reply:
Why not remove it if it is always the same value?

@jsimnz do you have a plan for this prop?

Reply:
it used to pass the same value, but now it's used for indexes to fetch a document for which an index needs to be updated. If the doc has bunch of props, this can play significant role (given the fetcher takes advantage of it)

Reply:
> Why not remove it if it is always the same value?
> @jsimnz do you have a plan for this prop?

Current dev branch doesn't do anything with the `[]FieldDescription`, regardless of the value, it will always return all fields for the collection. The pending Fetcher PR *does* make use of this field, and will respect the defined set of fields by only returning those. If `nil` is passed, it will return *all* fields. 

Since this will be merged pretty soon, its prob best to keep this here on this PR.

Reply:
cheers